src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs

```diff
@@ -435,14 +435,14 @@ private void OnStartEditingCellType(string name, bool switchTab)
             return;
         }
 
-        var newTypeToEdit = EditedSpecies.CellTypes.First(c => c.TypeName == name);
+        var newTypeToEdit = EditedSpecies.CellTypes.FirstOrDefault(c => c.TypeName == name);
```

Comment:
Why is this change required? If the new type to edit is not found shouldn't the code print an error, clear the cell type to be edited and then skip the rest of the code?

Reply:
Using FirstOrDefault saves us from having to handle a null selection as a special case. If I leave that as just First, it'll throw an exception when we emit OnCellTypeToEditSelected with a null or empty name. Until now, we never actually cleared the selection once it was set, so deleted cell types could still be seen in the editor.

Reply:
See my later comment, I did eventually discover this situation. I'd prefer an explicit check against null at the start of this method which would act as a special value meaning "clear selection" instead of try to pick a cell type like that.